lib/features/manage_account/presentation/profiles/identities/identities_controller.dart

```diff
@@ -287,8 +306,45 @@ class IdentitiesController extends BaseController {
         AppLocalizations.of(currentContext!).you_are_changed_your_identity_successfully);
     }
 
+    _cleanUpPublicAssets(success.identityId);
+
     _refreshAllIdentities();
   }
 
   bool get isSignatureShow => identitySelected.value != null;
+
+  void _injectCleanUpPublicAssetsInteractorBindings(Session? session, AccountId? accountId) {
+    try {
+      requireCapability(session!, accountId!, [CapabilityIdentifier.jmapPublicAsset]);
+      CleanUpPublicAssetsInteractorBindings().dependencies();
+    } catch(e) {
+      logError('$runtimeType::injectCleanUpPublicAssetsInteractorBindings(): exception: $e');
+    }
+  }
+
+  void _cleanUpPublicAssets(IdentityId? identityId) {
+    final session = accountDashBoardController.sessionCurrent;
+    final accountId = accountDashBoardController.accountId.value;
+
+    if (session == null
+      || accountId == null
+      || identityId == null
+      || _publicAssetsInIdentityArguments == null) return;
+
+    _injectCleanUpPublicAssetsInteractorBindings(session, accountId);
+    
+    final cleanUpPublicAssetsInteractor = getBinding<CleanUpPublicAssetsInteractor>();
+    if (cleanUpPublicAssetsInteractor == null) return;
+    consumeState(cleanUpPublicAssetsInteractor.execute(
+      session,
+      accountId,
+      identityId: identityId,
+      publicAssetsInIdentityArguments: _publicAssetsInIdentityArguments!));
+  }
+
+  @override
+  void onClose() {
+    CleanUpPublicAssetsInteractorBindings().close();
```

Comment:
IMO, Only `inject` and `close dependency` once for consistency

Reply:
Done